src/client/push-client.js

```diff
@@ -143,35 +143,44 @@ export default class PushClient extends EventDispatch {
    * @return {Promise<PushSubscription>} A Promise that
    *  resolves with a PushSubscription if successful.
    */
-  async subscribe() {
+  subscribe() {
     // Check for permission
-    const permissionStatus = await this.requestPermission(false);
+    return this.requestPermission(false)
+    .then(permissionStatus => {
+      if (permissionStatus !== 'granted') {
+        this._dispatchStatusUpdate();
+        throw new SubscriptionFailedError(permissionStatus);
+      }
 
-    if (permissionStatus !== 'granted') {
-      this._dispatchStatusUpdate();
-      throw new SubscriptionFailedError(permissionStatus);
-    }
+      this.dispatchEvent(new PushClientEvent('requestingsubscription'));
 
-    this.dispatchEvent(new PushClientEvent('requestingsubscription'));
+      // Make sure we have a service worker and subscribe for push
+      return navigator.serviceWorker.register(this._workerUrl, {
+        scope: this._scope
+      });
+    })
+    .then(registration => {
```

Comment:
If `registrationReady` isn't used elsewhere you can change it to return the registration, letting you change these 6 lines to just 
```
.then(registrationReady)
```

Otherwise, you can at least change from
```
.then(() => {
  return registration;
})
```
to
```
.then(() => registration)
```